src/adhocracy_core/adhocracy_core/scripts/auto_transition_process_workflow.py

```diff
@@ -0,0 +1,109 @@
+"""Automatically transition workflow states.
+
+This is registered as console script.
+"""
+import transaction
+import argparse
+import inspect
+import logging
+
+from pyramid.paster import bootstrap
+from pyramid.registry import Registry
+from substanced.util import find_service
+
+from adhocracy_core.authorization import create_fake_god_request
+from adhocracy_core.interfaces import IResource
+from adhocracy_core.interfaces import search_query
+from adhocracy_core.resources.process import IProcess
+from adhocracy_core.sheets.workflow import IWorkflowAssignment
+from adhocracy_core.workflows import transition_to_states
+from adhocracy_core.utils import now
+
+
+logger = logging.getLogger(__name__)
+
+
+def auto_transition_process_workflow():  # pragma: no cover
+    """Automatically transition workflow states.
+
+    usage::
+
+        bin/auto_auto_transition_process_workflow etc/development.ini
+    """
+    docstring = inspect.getdoc(auto_transition_process_workflow)
+    parser = argparse.ArgumentParser(description=docstring)
+    parser.add_argument('ini_file',
+                        help='path to the adhocracy backend ini file')
+    args = parser.parse_args()
+    env = bootstrap(args.ini_file)
+    _auto_transition_process_workflow(env['root'], env['registry'])
+    transaction.commit()
+    env['closer']()
+
+
+def _auto_transition_process_workflow(context: IResource, registry: Registry):
+    catalogs = find_service(context, 'catalogs')
+    query = search_query._replace(interfaces=IProcess,
+                                  resolve=True
```

Comment:
The resolve option return a list instead of a generator,  to decrease memory consumption we should always prefere to work with generators.